### PR TITLE
Issue #438 Fix the interceptor ordering test for both CDI Lite and Full

### LINKED ISSUE #438: org.jboss.cdi.tck.tests.interceptors.definition.interceptorOrder.InterceptorOrderTest
Due to the `cdi-full` TCK separation, this test is now invalid:

1. `testInterceptorsCalledInOrderDefinedByBeansXml` actually verifies `@Priority` ordering,
2. `testInterceptorsInvocationOrder` uses `@AroundInvoke` interceptors declared on target beans, which doesn't have to be supported by CDI Lite implementations.

I propose to:

1. Create a copy in the `full` package (with the `cdi-full` category) that restores the original test, including the `beans.xml` file.
2. In the current test, remove the `testInterceptorsInvocationOrder` method and then rename `testInterceptorsCalledInOrderDefinedByBeansXml` to `testInterceptorsInvocationOrder`.

### PR DESCRIPTION
**Pull request by Ladicek**
Fixes #438

### CODE REVIEW

File: impl/src/main/resources/tck-audit-cdi.xml [hunk 1]
```diff
@@ -5157,6 +5157,13 @@
             </text>
         </assertion>
 
+        <assertion id="l">
+            <text>The Priority annotation can be used to enable and order interceptors associated with components that use interceptor bindings.
```
Comment (manovotn) [inline impl/src/main/resources/tck-audit-cdi.xml:5161]: I don't mind this, just FTR - there is a `tck-audit-int` file which will probably already have a section related to this.

**Review by manovotn — APPROVED** [review]

File: impl/src/main/resources/tck-audit-cdi.xml (hunk 1 above)
Comment (Ladicek) [reply impl/src/main/resources/tck-audit-cdi.xml:5161]: `tck-audit-int` has an `id-`less assertion in `priority_annotation_in_ordering_int` section, with a note that "Tested within the CDI TCK, section 9.4 Interceptor enablement and ordering", so I'm not entirely sure: should I quote this assertion from `tck-audit-int` instead? Adding a new assertion to the "Interceptor enablement and ordering" section of `tck-audit-cdi` seemed a bit better to me, but I'm not really sure.

Comment (manovotn) [reply impl/src/main/resources/tck-audit-cdi.xml:5161]: I see In that case, the current solution seems good to me

**Comment by jeanouii** [comment]
Looks good

**Comment by Emily-Jiang** [comment]
In light of recent TCK runs, it was evident that this fix has caused a previous compatible implementation no longer passes this test. According to the [Jakarta TCK process](https://jakarta.ee/committees/specification/tckprocess/), the accepted changes normally just exclude tests or expanding tests but not completely changing tests. 

> **Accepted Challenges**
> 
> A consensus that a test produces invalid results will result in the exclusion of that test from certification requirements, and an immediate update and release of an official distribution of the TCK including the new exclude list. The associated challenge issue MUST be closed with an accepted label to indicate it has been resolved.
> 
> The specification project may approve (user) workarounds for an accepted TCK challenge (as an alternative to excluding TCK tests).
> 
> As another alternative to excluding a challenged test, it may be possible to adjust the test validation logic to “expand” the validation check. E.g. if a test expects a value “A1” for a specific variable “x”, but a challenge is raised arguing that the specification language actually allows for either values “A1” and “A2” (but no other values) to be valid values for “x”, then it could be a valid course of action to modify the challenged test to allow either “A1” OR “A2” for “x”.
> 
> Since this line of thinking might be applied to cases that aren’t quite as straightforward as this example, care should be taken when using this approach. A particular danger is that an implementation that has already demonstrated compliance before the challenge was raised might actually not pass the new, modified test.
> 
> To limit the confusion and additional work such a scenario would cause, if there is already at least one certified compatible implementation before the challenge, the new, modified TCK should be run against at least one such implementation (and ideally all of them) before the changes are published, released, and finalized.
> 
> If such a change were released, and it was later found to cause a previously-certified implementation to fail the new, modified test, then excluding the test would likely be the only option, and this would require yet another, additional service release.
If this PR will be released under a major or minor release, it is fine. However, it does not comply with the TCK process if it is made to a service release.

**Comment by Ladicek** [comment]
I believe your comment belongs to #431 and not here. Correct?

**Comment by Emily-Jiang** [comment]
> I believe your comment belongs to #431 and not here. Correct?

oops. you were right. sorry about this.

**Comment by Ladicek** [comment]
No worries, I was just wondering, because I'm pretty sure I verified this PR with Weld prior to submitting.

**Pull request merged**